From stable+bounces-294160-greg=kroah.com@vger.kernel.org Fri Jul 31 19:15:48 2026
From: Sasha Levin <sashal@kernel.org>
Date: Fri, 31 Jul 2026 12:56:55 -0400
Subject: drm/xe: Move xe->info.force_execlist initialization
To: stable@vger.kernel.org
Cc: Michal Wajdeczko <michal.wajdeczko@intel.com>, Gustavo Sousa <gustavo.sousa@intel.com>, Sasha Levin <sashal@kernel.org>
Message-ID: <20260731165658.883265-2-sashal@kernel.org>

From: Michal Wajdeczko <michal.wajdeczko@intel.com>

[ Upstream commit f1d581bb50ed54b71a8121d3d46fe2627fddd147 ]

The xe_info_init_early() is a place where we initialize those of
the xe->info fields that do not require any additional hardware
probes. Move the initialization of the force_execlist flag there.

Signed-off-by: Michal Wajdeczko <michal.wajdeczko@intel.com>
Reviewed-by: Gustavo Sousa <gustavo.sousa@intel.com>
Link: https://patch.msgid.link/20260526195452.20545-4-michal.wajdeczko@intel.com
Stable-dep-of: ba7fd1634228 ("drm/xe: Set TTM device beneficial_order to 9 (2M)")
Signed-off-by: Sasha Levin <sashal@kernel.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---
 drivers/gpu/drm/xe/xe_device.c |    1 -
 drivers/gpu/drm/xe/xe_pci.c    |    1 +
 2 files changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/gpu/drm/xe/xe_device.c
+++ b/drivers/gpu/drm/xe/xe_device.c
@@ -475,7 +475,6 @@ struct xe_device *xe_device_create(struc
 
 	xe->info.devid = pdev->device;
 	xe->info.revid = pdev->revision;
-	xe->info.force_execlist = xe_modparam.force_execlist;
 	xe->atomic_svm_timeslice_ms = 5;
 	xe->min_run_period_lr_ms = 5;
 
--- a/drivers/gpu/drm/xe/xe_pci.c
+++ b/drivers/gpu/drm/xe/xe_pci.c
@@ -774,6 +774,7 @@ static int xe_info_init_early(struct xe_
 	xe->info.probe_display = IS_ENABLED(CONFIG_DRM_XE_DISPLAY) &&
 				 xe_modparam.probe_display &&
 				 desc->has_display;
+	xe->info.force_execlist = xe_modparam.force_execlist;
 
 	xe_assert(xe, desc->max_gt_per_tile > 0);
 	xe_assert(xe, desc->max_gt_per_tile <= XE_MAX_GT_PER_TILE);
